core/orchestrator.go

```diff
@@ -563,9 +568,15 @@ func (rtm *RemoteTranscoderManager) Transcode(fname string, profiles []ffmpeg.Vi
 	}
 	res, err := currentTranscoder.Transcode(fname, profiles)
 	_, fatal := err.(RemoteTranscoderFatalError)
-	if !fatal {
-		rtm.completeTranscoders(currentTranscoder)
+	if fatal {
+		// Don't retry if we've timed out; broadcaster likely to have moved on
```

Comment:
Interesting! So we're not retrying here because the B probably has already sent the segment that's taking too long to another O / T pair for transcoding, so a retry would result in double transcoding? Is that kind of redundancy here a big issue?

Reply:
Correct, it's a "selfish" optimization to save the O from doing unnecessary work if it's likely the B isn't depending on its results any longer. Don't think it's a big issue, but it was easy to add in. We might have to address the issue for VOD though, depending on how we do it.

Reply:
Makes sense. I see how multi-T/O architecture wouldn't be very important for VOD, since we don't need real-time transcoding and can retry using the same O / T as many times as we wanted (unless there's a failure)